### PR TITLE
Replace mkdirp with own implementation

### PR DESCRIPTION
**Pull request by lucleray**
Same idea like in https://github.com/zeit/next.js/issues/6941

This PR replaces `mkdirp` node module by a simpler/lighter implementation. It reduces install size of 40kB+ according to packagephobia.

Node v10.12.0 introduces a `recursive` option to [`fs.mkdir`](https://nodejs.org/api/fs.html#fs_fs_mkdir_path_options_callback) and we can use it when we drop node 8 support.

### CODE REVIEW

**Review by lfades — APPROVED** [review]

**Comment by timneutkens** [comment]
Could you do a benchmark similar to the `bench` directory?

File: bench/mkdirp/run.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+const { join } = require('path')
+const mkdirpNpm = require('mkdirp')
+const { mkdirp: mkdirpCustom } = require('mkdirp')
```
Comment (ijjk) [inline bench/mkdirp/run.js]:
```suggestion
const { mkdirp: mkdirpCustom } = require('next/dist/lib/mkdirp')
```

**Comment by lucleray** [comment]
@timneutkens 

> Could you do a benchmark similar to the `bench` directory?

I added a benchmark in `bench/recursive-copy`.

**Comment by timneutkens** [comment]
@lucleray could you share the results?

**Comment by lucleray** [comment]
@timneutkens 

```
~ cd bench
~ yarn run bench:recursive-copy
test recursive-copy npm module
{ sum: 1067.209105, nb: 10, avg: 106.72091049999999 }
test recursive-copy custom implementation
{ sum: 582.435537, nb: 10, avg: 58.24355369999999 }
```

**Comment by timneutkens** [comment]
So if we can't make it faster let's maybe `ncc` the `mkdirp` module 👍

**Comment by lucleray** [comment]
I added a piece of code to detect node version and, when available use `mkdir(dir, {recursive:true})`

`mkdirp` ends up being faster on node >= v10.12 :

```
luc ~/w/n/n/bench ‣ nvm use lts/latest
luc ~/w/n/n/bench ‣ yarn run bench:mkdirp
yarn run v1.15.2
warning package.json: No license field
$ node mkdirp/run.js
test mkdirp npm module
{ sum: 177.70820999999998, nb: 100, avg: 1.7770820999999999 }
test mkdirp custom implementation
{ sum: 142.06693600000006, nb: 100, avg: 1.4206693600000007 }
test mkdirp sync npm module
{ sum: 162.313997, nb: 100, avg: 1.62313997 }
test mkdirp sync custom implementation
{ sum: 115.01636, nb: 100, avg: 1.1501636 }
✨  Done in 1.27s.
```

**Comment by Timer** [comment]
@lucleray since this makes the install size bigger let's postpone this until January of next year when Node 8 is EOL (and we can only use the built-in version). 🙏 Thank you for all your efforts!

**Comment by lucleray** [comment]
@Timer I forgot to remove `mkdirp` from `packages/next/package.json` hence the bigger size. I'll do it now.

**Pull request closed without merging**